### PR TITLE
SDKS-5265 Build the ping_oidc bridge package

### PR DESCRIPTION
**Pull request by rodrigoareis**
###  Summary

 Adds `ping_oidc`, a new Flutter bridge plugin package exposing the native Ping OIDC SDK (Android/iOS) to Dart, following the same Pigeon-generated architecture established by ping_journey in #124.

  - **flutter/flutter-sdk-bridge/ping_oidc/** — the new package:
    - Pigeon schema (pigeons/messages.dart) and generated Dart/Kotlin/Swift bindings
    - `OidcClientFactory` / `OidcConfigParser` / `OidcHostApiImpl` / `OidcWebClientFactory` on both platforms, wrapping the native `OidcClient` and `OidcWebClient` behind a two-handle
  model (config handle + browser-capable web-client handle)
    - Typed error mapper (`OidcErrorMapper`/`OidcErrorCodes`) kept 1:1 between Kotlin and Swift
    - Dart facade `OidcClient` exposing `configure`, `authorize`, `hasUser`, `token`, `refresh`, `userInfo`, `revoke`, `signOff`, and `dispose`
    - Pinned to native Ping SDK 2.1.0 on both platforms (Android Maven, iOS SPM)
  - **flutter/flutter-sdk-bridge/ping_core/** — small supporting additions:
    - OidcConfigHandle / OidcOpenIdConfig (Dart/Kotlin/Swift) — shared config plumbing for the not-yet-built Journey/OIDC delegation seam
    - Relocated `JsonBridgeMapper` from `ping_journey` into `ping_core` (mechanical move, no behavior change)
  - Small supporting tweaks to ping_journey (`JourneyHostApiImpl.kt`, `JourneyNodeMapper.kt`) and version bumps across READMEs/podspecs/Package.swift files.

###  Test plan

  - [x] flutter test passes for ping_core and ping_oidc (unit tests)
  - [x] flutter analyze clean for both packages
  - [ ] Manual OIDC flow verification against a configured PingOne/PingAM tenant (login, token/userinfo retrieval, refresh, revoke, sign-off) — deferred to the flutter-oidc
  sample app in SDKS-5266
  - [x] Android and iOS both build

### CODE REVIEW

File: flutter/flutter-sdk-bridge/ping_oidc/ios/ping_oidc/Sources/ping_oidc/OidcHostApiImpl.swift [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    /// TODO(SDKS-5295): ping-ios-sdk's `OidcWebClient.authorize()` collapses every
+    /// `FailureNode.cause` into `OidcError.unknown(message:)`, discarding
+    /// `BrowserError.externalUserAgentCancelled`'s type entirely (confirmed via source trace —
+    /// `Web.swift`'s bare `catch` rewraps it as a string before it ever reaches here). This
+    /// substring match on the resulting message is the only signal available today. It is
+    /// fragile — it silently stops working if the SDK ever rewords the message — and should be
+    /// removed once the native SDK preserves a distinguishable cancellation signal, tracked in
+    /// SDKS-5295 to align it with Android's typed `BrowserCanceledException`.
+    private func mapAuthorizeFailure(_ error: OidcError) -> Result<AuthorizeResultMessage, Error> {
+        if plainErrorMessage(for: error).contains("cancelled by the user") {
```
Comment (george-bafaloukas-forgerock) [inline flutter/flutter-sdk-bridge/ping_oidc/ios/ping_oidc/Sources/ping_oidc/OidcHostApiImpl.swift:82]: Nice catch on documenting the cancellation-detection asymmetry here — the TODO linking SDKS-5295 is genuinely helpful for whoever picks that up.

One thing worth flagging: this substring match (`"cancelled by the user"`) is the only signal available today because `ping-ios-sdk`'s `OidcWebClient.authorize()` collapses `BrowserError.externalUserAgentCancelled` into a generic `OidcError.unknown(message:)`, so it's a real (if well-documented) production risk rather than an implementation oversight. If `ping-ios-sdk` ever rewords or localizes that message, this will silently misclassify cancellations as errors — and no test in this PR would catch that regression since it depends on upstream SDK internals not exercised here. Might be worth a comment pointing at this exact string as the trip-wire to check whenever `ping-ios-sdk` is bumped, until SDKS-5295 lands a typed signal.

File: flutter/flutter-sdk-bridge/ping_oidc/android/src/main/kotlin/com/pingidentity/flutter/oidc/OidcHostApiImpl.kt [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+import kotlinx.coroutines.launch
+import com.pingidentity.utils.Result as OidcResult
+
+/**
+ * Implements the generated [PingOidcHostApi]. `configureOidc`/`dispose` (Phase 2),
+ * `createWebClient`/`authorize`/`hasUser` (Phase 3), and `token`/`refresh`/`userInfo`/`revoke`/
+ * `signOff` (Phase 4) all have real bodies now.
+ */
+class OidcHostApiImpl : PingOidcHostApi {
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
```
Comment (george-bafaloukas-forgerock) [inline flutter/flutter-sdk-bridge/ping_oidc/android/src/main/kotlin/com/pingidentity/flutter/oidc/OidcHostApiImpl.kt:32]: Small suggestion: the `CoroutineScope(SupervisorJob() + Dispatchers.IO)` here is correctly cancelled today via `PingOidcPlugin.onDetachedFromEngine`'s `shutdown()` call, but there's no lifecycle test exercising attach/detach/re-attach. If a future refactor drops or reorders that `shutdown()` call, stale-instance callbacks could fire after replacement (e.g. on Android configuration changes or multi-engine apps) with nothing catching the regression. Might be worth a lifecycle test if this plugin is expected to survive engine detach/reattach — not blocking, just a gap worth being aware of.

File: flutter/flutter-sdk-bridge/ping_oidc/lib/src/oidc_client.dart [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    );
+    final String webHandleId;
+    try {
+      webHandleId = await _guard(
+        () => api.createWebClient(handleId, browserOptions?.toMessage()),
+      );
+    } catch (error) {
+      // Roll back the already-registered client handle so a createWebClient failure can't
+      // leave it orphaned in CoreRuntime.oidcClientRegistry with no caller-reachable dispose().
+      await _guard(() => api.dispose(handleId));
```
Comment (george-bafaloukas-forgerock) [inline flutter/flutter-sdk-bridge/ping_oidc/lib/src/oidc_client.dart]: Nice touch adding rollback here on `createWebClient` failure — disposing the already-registered client handle so we don't leak on partial failure, and it looks well covered by tests.

One small edge case worth a mental note: this is a "best effort" rollback — if `dispose(handleId)` inside the `catch` itself throws, `_guard`'s `rethrow` correctly surfaces the *original* error rather than swallowing it, which is the right call. But there's no log/telemetry hook for that secondary failure, so a double-failure case (original error + failed rollback) would leak a handle silently in production with no visibility into it. Not blocking, just something to keep in mind if handle leaks ever need to be diagnosed in the field.

**Review by spetrov — CHANGES_REQUESTED** [review]
## Build / test configuration issues found during review

The following issues prevent the native unit tests from being run out-of-the-box.
All are fixable; flagging so they can be addressed before the sample app (SDKS-5266) ships.

### Android — `ping_oidc/android` and `ping_core/android`

  1. **`settings.gradle.kts` missing `pluginManagement` block**
     
Opening either `android/` directory standalone in Android Studio fails immediately because the `com.android.library` plugin cannot be located.
     
**Fix:** Add a `pluginManagement { repositories { google(); mavenCentral(); gradlePluginPortal() } }` block to both `settings.gradle.kts` files.

  2. **`id("org.jetbrains.kotlin.android")` incompatible with AGP 9.0**
     
AGP 9.0 bundles Kotlin natively. Declaring the standalone Kotlin Android plugin alongside it produces a hard error during configuration.

**Fix:** Remove `id("org.jetbrains.kotlin.android")` from `plugins {}` in both `build.gradle.kts` files (and from `pluginManagement.plugins` if present).

  3. **`flutter.jar` not on the compile classpath for standalone builds**
     
`PingCorePlugin.kt` imports `io.flutter.embedding.engine.plugins.FlutterPlugin`.
Flutter injects `flutter.jar` automatically when it drives the build, but not when Android Studio opens `android/` directly.

**Fix:** Read `flutter.sdk` from `local.properties` (or `$FLUTTER_ROOT`) and add `compileOnly(fileTree("$flutterSdk/bin/cache/artifacts/engine/android-arm", 
include: ["flutter.jar"]))`. Document that `local.properties` must be created pointing to the Flutter SDK installation.

  4. **Wrong test dependency — `kotlin-test` instead of `kotlin-test-junit5`**
     
`useJUnitPlatform()` is active but `kotlin-test` does not pull in the JUnit 5 bridge, causing `@Test` to be unresolved in test source.

**Fix:** Replace `kotlin-test` with `kotlin-test-junit5` and add `junit-jupiter:5.11.4` + `junit-platform-launcher` (testRuntimeOnly).

  5. **Ping SDK 2.x artifacts compiled for Java 21; tests fail with `UnsupportedClassVersionError` on Java 17**
     
The test JVM must be Java 21 or later. Running `./gradlew test` with a Java 17 JDK causes every test that touches native SDK classes to fail at class-load time.
     
**Fix:** Add `java { toolchain { languageVersion.set(JavaLanguageVersion.of(21)) } }` to both `build.gradle.kts` files.

### iOS — `ping_oidc/ios` and `ping_core/ios`

  6. **Swift tests cannot be run standalone (neither `swift test` nor direct Xcode)**
     
`Package.swift` declares `../FlutterFramework` as a local path dependency that Flutter generates only when building a host app. On a clean checkout that directory is absent, so SPM cannot resolve the package graph and Xcode cannot present the test scheme. Even with the stub present, `PingOidcPlugin.swift` and `Messages.g.swift` unconditionally import `Flutter` and `UIKit`, which are only available inside a Flutter app's Xcode workspace (`Runner.xcworkspace`). Since this PR does not include a sample app, the Swift test targets are currently unreachable.


**Fix options:** 
     **(a)** Add an `example/` app (standard Flutter plugin pattern) that provides the Xcode workspace context
      or...
     **(b)** guard platform-specific imports with `#if canImport(Flutter)` and add macOS/Linux stubs so `swift test` can at least exercise the pure-logic tests on the macOS host.

**Comment by rodrigoareis** [comment]
@spetrov Thanks for the detailed review, I reviewed each item and pushed a new commit with fixes. 

- Regarding item 6,  `Swift tests unreachable`, it's confirmed, and broader than described: not unique to `ping_oidc`. `FlutterFramework` has never existed anywhere in this repo, so no Swift bridge test has ever run, on this PR or the previous merged. We looked at guarding the plugin class with `#if canImport(Flutter)`, but it doesn't help — `Messages.g.swift` (Pigeon-generated) is load-bearing for the `error-mapping/config-parsing` code too and unconditionally needs a Flutter/FlutterMacOS module. A real fix needs your option (a), an example/ app — deferring to SDKS-5266, which will finally provide that context.

- Two more things this surfaced:
  - Fixing item 5 alone unmasked a mockito-core:5.0.0/ByteBuddy incompatibility on JDK 21. Bumped to 5.14.2 in all three packages.
  - CI itself was silently broken, unrelated to any of this: flutter test ran from the workspace root (no test/ dir there), failing before ever reaching the Android step you reviewed — on this PR and previous both. Fixed the step to iterate per package, and added a new CI step for `ping_oidc`'s own tests (it had zero CI coverage before).

File: flutter/flutter-sdk-bridge/ping_oidc/ios/ping_oidc/Sources/ping_oidc/OidcHostApiImpl.swift (hunk 1 above)
Comment (rodrigoareis) [reply flutter/flutter-sdk-bridge/ping_oidc/ios/ping_oidc/Sources/ping_oidc/OidcHostApiImpl.swift:82]: Good catch — added a comment above the check naming this exact string as the trip-wire to verify on any `ping-ios-sdk` bump, until `SDKS-5295` lands a typed signal.

File: flutter/flutter-sdk-bridge/ping_oidc/android/src/main/kotlin/com/pingidentity/flutter/oidc/OidcHostApiImpl.kt (hunk 2 above)
Comment (rodrigoareis) [reply flutter/flutter-sdk-bridge/ping_oidc/android/src/main/kotlin/com/pingidentity/flutter/oidc/OidcHostApiImpl.kt:32]: Added a test exercising exactly this — asserts a call issued after `shutdown()` never invokes its callback, so a future refactor that drops the `shutdown()` call now has a regression test to catch it.

File: flutter/flutter-sdk-bridge/ping_oidc/lib/src/oidc_client.dart (hunk 3 above)
Comment (rodrigoareis) [reply flutter/flutter-sdk-bridge/ping_oidc/lib/src/oidc_client.dart]: Added a `debugPrint` in the rollback `catch` naming the handle id and the rollback error, so a leaked-handle report has something to point at. Still rethrows the original error.

**Review by spetrov — APPROVED** [review]
